### PR TITLE
Fix #179 : alias for upsert calls

### LINKED ISSUE #179: alias upsert => upsert_one (with deprecation)
out of least-surprise, cf. insert_one and insert_many for instance.

### PR DESCRIPTION
**Pull request by erichare**
deprecate upsert(), use upsert_one() instead

### CODE REVIEW

File: astrapy/db.py [hunk 1]
```diff
@@ -1753,6 +1761,14 @@ async def delete_subdocument(self, id: str, subdoc: str) -> API_RESPONSE:
         return response
 
     async def upsert(self, document: API_DOC) -> str:
+        # Deprecated: call the upsert_one method
+        DEPRECATION_MESSAGE = "Method 'upsert' of AstraDBCollection is deprecated. Please switch to method 'upsert_one'."
+
+        warn(DEPRECATION_MESSAGE, DeprecationWarning, stacklevel=2)
```
Comment (cbornet) [inline astrapy/db.py:1767]: The method `logging.warn` is deprecated. `logging.warning` should be used instead.

Comment (cbornet) [inline astrapy/db.py:1767]: We could factorize in a `deprecation_warning` method ?

Comment (hemidactylus) [reply astrapy/db.py:1767]: This is `from warnings import warn` being used, probably good as is (no?)

Comment (hemidactylus) [reply astrapy/db.py:1767]: Agree- maybe a method decorator is even less intrusive. For that matter we might just use [this](https://pypi.org/project/deprecation/).

**Review by hemidactylus — APPROVED** [review]

File: astrapy/db.py (hunk 1 above)
Comment (erichare) [reply astrapy/db.py]: I like this idea! I'm going to push an update now which uses the deprecation module. it looks very dev friendly, and seems to work with our tests. We can easily update the messages etc and now they're just decorators on the function itself. See the update and let me know if you guys have any further feedback @cbornet @hemidactylus . Thanks :)

Comment (cbornet) [reply astrapy/db.py]: Oh, sorry for the noise. It’s logging.warn which is deprecated…

**Pull request merged**